manifests/repo.pp

```diff
@@ -46,64 +46,74 @@
-        }
+    'deb': {
+      $osname = downcase($::operatingsystem)
+      case $osname {
+        'debian', 'ubuntu': {
+
+          $component = 'main'
+          $repo_url = "${base_url}/${repo_name}/${osname}"
+          $distribution =  $::lsbdistcodename
+
+          file { $normalized_name:
+            ensure  => file,
+            path    => "/etc/apt/sources.list.d/${normalized_name}.list",
+            mode    => '0644',
+            content => template('packagecloud/apt.erb'),
+          }
 
-        exec { "apt_key_add_${normalized_name}":
-          command => "wget --auth-no-challenge -qO - ${base_url}/${repo_name}/gpgkey | apt-key add -",
-          path    => '/usr/bin/:/bin/',
-          require => File[$normalized_name],
-        }
+          exec { "apt_key_add_${normalized_name}":
+            command => "wget --auth-no-challenge -qO - ${base_url}/${repo_name}/gpgkey | apt-key add -",
+            path    => '/usr/bin/:/bin/',
+            require => File[$normalized_name],
+          }
 
-        exec { "apt_get_update_${normalized_name}":
-          command =>  "apt-get update -o Dir::Etc::sourcelist=\"sources.list.d/${normalized_name}.list\" -o Dir::Etc::sourceparts=\"-\" -o APT::Get::List-Cleanup=\"0\"",
-          path    => '/usr/bin/:/bin/',
-          require => Exec["apt_key_add_${normalized_name}"],
+          exec { "apt_get_update_${normalized_name}":
+            command =>  "apt-get update -o Dir::Etc::sourcelist=\"sources.list.d/${normalized_name}.list\" -o Dir::Etc::sourceparts=\"-\" -o APT::Get::List-Cleanup=\"0\"",
+            path    => '/usr/bin/:/bin/',
+            require => Exec["apt_key_add_${normalized_name}"],
+          }
+        }
+        default: {
+          fail("Sorry, ${::operatingsystem} isn't supported for apt repos at this time. Email support@packagecloud.io")
         }
-      }
-
-      default: {
-        fail("Sorry, ${::operatingsystem} isn't supported for apt repos at this time. Email support@packagecloud.io")
       }
     }
-  } elsif $type == 'rpm' {
-    case $::operatingsystem {
-      'RedHat', 'redhat', 'CentOS', 'centos', 'Amazon', 'Fedora', 'Scientific', 'OracleLinux', 'OEL': {
-
-        $majrel = $::osreleasemaj
-        if $::pygpgme_installed == 'false' {
-          warning('The pygpgme package could not be installed. This means GPG verification is not possible for any RPM installed on your system. To fix this, add a repository with pygpgme. Usualy, the EPEL repository for your system will have this. More information: https://fedoraproject.org/wiki/EPEL#How_can_I_use_these_extra_packages.3F and https://github.com/stahnma/puppet-module-epel')
-          $repo_gpgcheck = 0
-        } else {
-          $repo_gpgcheck = 1
-        }
+    'rpm': {
+      case $::operatingsystem {
+        'RedHat', 'redhat', 'CentOS', 'centos', 'Amazon', 'Fedora', 'Scientific', 'OracleLinux', 'OEL': {
+
+          $majrel = $::osreleasemaj
+          if $::pygpgme_installed == false {
```

Comment:
ah, yes, this is because `$::pygpgme_installed` is a string, not a truthy value. looks like in the refactor it got switched to a truthy. there was a reason why i made `$::pygpgme_installed` be a string instead of a truthy originally. it was something about facter functions, but i can't remember why.

if you commit a change to change this back to a string comparison, that should fix the spec and i can merge this.